src/ConfigParser.cpp

```diff
@@ -1,14 +1,103 @@
 #include "../include/ConfigParser.hpp"
+#include "../include/Server.hpp"
+#include "ConfigParser.hpp"
 
-//ConfigParser::ConfigParser() : lineCount(0), start(0)
-//{
-//}
+void ConfigParser::parseConfig(const std::string& filename) 
+{
+	std::ifstream file(filename);
+	std::string line;
 
-//ConfigParser::~ConfigParser()
-//{
-//}
+	if (!file.is_open()) 
+	{
+		std::cerr << "Failed to open file: " << filename << std::endl;
+		return;
+	}
 
-//const char	*ConfigParser::ErrorEnviron::what() const throw()
-//{
-//	return ("Webserv: Cannot find path");
-//}
+	std::string currentSection;
+	while (std::getline(file, line)) 
+	{
+		// Ignore comments
+		size_t commentPos = line.find('#');
+		if (commentPos != std::string::npos) 
+			line = line.substr(0, commentPos);
+
+		// Trim whitespace
+		line = trim(line);
+
+		if (!line.empty()) 
+		{
+			if (line == "{") 
+				sectionStack.push_back(currentSection); 
+			else if (line == "}") 
+			{
+				if (!sectionStack.empty()) 
+				{
+					sectionStack.pop_back();
+					if (!sectionStack.empty()) 
+						currentSection = sectionStack.back();
+					else 
+						currentSection.clear();
+				}
+			} 
+			else 
+			{
+				processLine(line);
+			}
+		}
+	}
+}
+
+const std::vector<std::shared_ptr<Server> > &ConfigParser::getServers() const
+{
+	return servers;
+}
+
+void ConfigParser::processLine(const std::string &line)
+{
+	std::istringstream iss(line);
+	std::string keyword;
+	iss >> keyword;
+
+	if (keyword == "server") 
+	{
+		currentServer = std::make_shared<Server>();
+		servers.push_back(currentServer);
+	}
+	if (keyword == "main" || keyword == "location")
+	{
+		currentSection = keyword;
+		if (keyword == "location") 
+		{
+			std::string locationName;
+			iss >> locationName;
+			currentSection += " " + locationName;
+		}
+	}
+	if (currentSection == "main")
+	{
+		std::string value;
+		iss >> value;;
+		if (currentServer)
+		{
+			if (keyword == PARSEHOST)
```

Comment:
using string,compare function is safer than using the ==, so rather have `keyword.compare(PARSEHOST) == 0`